internal/host/plugin/repository_host_catalog_test.go

```diff
@@ -1441,6 +1444,136 @@ func TestRepository_DeleteCatalogX(t *testing.T) {
 	}
 }
 
+// Create a test scheduled job that we can use to catch when a run is executed.
+// We will use this to track whether or not an update was sent in
+// TestRepository_UpdateCatalog_SyncSets.
+type testSyncJob struct{}
+
+func (j *testSyncJob) Status() scheduler.JobStatus {
+	return scheduler.JobStatus{
+		Completed: 0,
+		Total:     1,
+	}
+}
+
+func (j *testSyncJob) Run(_ context.Context) error       { return nil }
+func (j *testSyncJob) NextRunIn() (time.Duration, error) { return setSyncJobRunInterval, nil }
+func (j *testSyncJob) Name() string                      { return setSyncJobName }
+func (j *testSyncJob) Description() string               { return setSyncJobName }
+
+func TestRepository_UpdateCatalog_SyncSets(t *testing.T) {
+	ctx := context.Background()
+	dbConn, _ := db.TestSetup(t, "postgres")
+	dbRW := db.New(dbConn)
+	dbWrapper := db.TestWrapper(t)
+	sched := scheduler.TestScheduler(t, dbConn, dbWrapper)
+	dbKmsCache := kms.TestKms(t, dbConn, dbWrapper)
+	_, projectScope := iam.TestScopes(t, iam.TestRepo(t, dbConn, dbWrapper))
+
+	testPlugin := hostplg.TestPlugin(t, dbConn, "test")
+	dummyPluginMap := map[string]plgpb.HostPluginServiceClient{
+		testPlugin.GetPublicId(): &WrappingPluginClient{Server: &plgpb.UnimplementedHostPluginServiceServer{}},
+	}
+
+	// Set up a test catalog and the secrets for it
+	testCatalog := TestCatalog(t, dbConn, projectScope.PublicId, testPlugin.GetPublicId())
+	// Set up another test catalog that will not have sets on it
+	emptyTestCatalog := TestCatalog(t, dbConn, projectScope.PublicId, testPlugin.GetPublicId())
+
+	// Create a couple of host sets.
+	set1 := TestSet(t, dbConn, dbKmsCache, sched, testCatalog, dummyPluginMap)
+	set1.LastSyncTime = timestamp.New(time.Now())
+	set1.NeedSync = false
+
+	numSetsUpdated, err := dbRW.Update(ctx, set1, []string{"LastSyncTime", "NeedSync"}, []string{})
+	require.NoError(t, err)
+	require.Equal(t, 1, numSetsUpdated)
+
+	// Set 2
+	set2 := TestSet(t, dbConn, dbKmsCache, sched, testCatalog, dummyPluginMap)
+	set2.LastSyncTime = timestamp.New(time.Now())
+	set2.NeedSync = false
+
+	numSetsUpdated, err = dbRW.Update(ctx, set2, []string{"LastSyncTime", "NeedSync"}, []string{})
+	require.NoError(t, err)
+	require.Equal(t, 1, numSetsUpdated)
+
+	j := &testSyncJob{}
+	err = sched.RegisterJob(ctx, j, scheduler.WithNextRunIn(setSyncJobRunInterval))
+	require.NoError(t, err)
+	var wg sync.WaitGroup
+	err = sched.Start(ctx, &wg)
```

Comment:
With the new way of validating the job's next run time was scheduled, you don't need to start the scheduler.

Reply:
Removed.